docs/multiplatform.md

```diff
@@ -0,0 +1,89 @@
+# Multiplatform
+
+This document covers using kotlin-inject in a multiplatform project.
+
+## Download
+
+Using [ksp](https://github.com/google/ksp)
+
+`settings.gradle`
+
+```kotlin
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+        mavenCentral()
+    }
+}
+
+dependencyResolutionManagement {
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
+```
+
+`build.gradle`
+
+Add Plugin
+```kotlin
+plugins {
+    id("org.jetbrains.kotlin.multiplatform") version "1.9.21"
+    id("com.android.library")
+    id("com.google.devtools.ksp") version "1.9.21-1.0.16"
+}
+```
+
+Add Runtime dependency in commonMain
+```kotlin
+sourceSets {
+    androidMain.dependencies {
+        //...
+    }
+    commonMain.dependencies {
+        implementation("me.tatarka.inject:kotlin-inject-runtime:0.6.3")
+    }
+}
+```
+If you're targetting iOS, you need to add each of the chipset as separate sourcesets to include generated files.
+Same goes to any other LLVM (MacOS, TVOS etc.) you're targetting.
+```kotlin
+val iosX64Main by getting {
+    dependencies {
+        dependsOn(iosMain.get())
+        kotlin.srcDir("build/generated/ksp/iosX64/iosX64Main/kotlin")
+    }
+}
+val iosArm64Main by getting {
+    dependencies {
+        dependsOn(iosMain.get())
+        kotlin.srcDir("build/generated/ksp/iosArm64/iosArm64Main/kotlin")
+    }
+}
+val iosSimulatorArm64Main by getting {
+    dependencies {
+        dependsOn(iosMain.get())
+        kotlin.srcDir("build/generated/ksp/iosSimulatorArm64/iosSimulatorArm64Main/kotlin")
+    }
+}
+```
+
+Add the compiler dependency in root level `dependencies` block for each of the platforms you're targetting
+```kotlin
+dependencies {
+    add("kspIosX64", "me.tatarka.inject:kotlin-inject-compiler-ksp:0.6.3")
+    add("kspIosArm64", "me.tatarka.inject:kotlin-inject-compiler-ksp:0.6.3")
+    add("kspIosSimulatorArm64", "me.tatarka.inject:kotlin-inject-compiler-ksp:0.6.3")
```

Comment:
we might need one for `kspCommonMainMetadata` as well I suppose?

Reply:
I didn't need that